packages/tasks_panel/src/misc/common-e2e-config.ts

```diff
@@ -31,20 +33,27 @@ export async function completeDeployConfig(task: any): Promise<void> {
 }
 
 export async function getConfigDeployPickItems(project: string): Promise<ProjectConfigInfo[]> {
-  const items = reduce(
-    await collectProjects(project),
-    (acc, info: ProjectInfo) => {
-      if (info.style === ProjTypes.FIORI_FE) {
-        acc.push(getFioriE2ePickItems(info));
-      } else if (info.style === ProjTypes.CAP) {
-        acc.push(getCapE2ePickItems(info));
-      } else if (info.style === ProjTypes.HANA) {
-        acc.push(getHanaE2ePickItems(info));
-      }
-      return acc;
-    },
-    [] as Promise<ProjectConfigInfo | undefined>[],
-  );
+  let items = [] as Promise<ProjectConfigInfo | undefined>[];
+  const requestedFolder = workspace.getWorkspaceFolder(Uri.file(project));
+  if (requestedFolder) {
+    const projects = filter(await collectProjects(requestedFolder.uri.path), (_) =>
+      isPathRelatedToFolder(join(_.wsFolder, _.project), project),
+    );
+    items = reduce(
```

Comment:
why did you use reduce and not just a simple loop collecting items?

Reply:
IMHO it more clear - the original collection of `items` can be mutated during the loop.